src/language-js/printer-estree.js

```diff
@@ -846,6 +847,32 @@ function canAttachComment(node) {
   );
 }
 
+function ensurePrintingNode(path) {
+  const parent = path.getParentNode();
+
+  if (!parent) {
+    return;
+  }
+
+  let name = path.getName();
+  if (typeof name === "number") {
+    name = path.stack[path.stack.length - 4];
```

Comment:
Can you add comment why `4`?

Reply:
https://github.com/prettier/prettier/pull/13293/commits/c94afaa01ee063a57406a4197c806c0d11af657c